File: src/types/validation.ts
export interface ConflictCandidate {
  candidate_id: string;
  value: string;
  unit?: string | null;
  source_name: string;
  source_type: 'pdf' | 'website' | 'catalog' | 'image' | 'third_party' | string;
  source_url?: string | null;
  page_number?: number | null;
  confidence: number;
  evidence_text: string;
}

export interface ConflictItem {
  id: string;
  product_id: string;
  product_name?: string;
  attribute_name: string;
  key: string;
  current_value?: string | null;
  status: 'conflict' | 'needs_review' | 'resolved' | string;
  candidates: ConflictCandidate[];
  created_at: string;
}

export interface ReviewActionRequest {
  action: 'select_candidate' | 'edit_manual' | 'mark_na' | 'reject_ai' | string;
  selected_candidate_id?: string;
  manual_value?: string;
  manual_unit?: string;
  reviewer_name?: string;
  notes?: string;
}

export interface ReviewHistoryItem {
  id: string;
  product_id: string;
  attribute_name: string;
  key: string;
  previous_value?: string | null;
  final_value: string;
  reviewer: string;
  action: string;
  timestamp: string;
}

export interface ValidationSummary {
  product_id: string;
  total_attributes_validated: number;
  matching_specs_count: number;
  conflicts_count: number;
  low_confidence_count: number;
  unverified_enriched_count: number;
  overall_confidence: number;
  confidence_tier: 'High Confidence' | 'Medium Confidence' | 'Low Confidence' | string;
  validation_status: 'verified' | 'human_verified' | 'needs_review' | 'conflict' | string;
}

File: backend/app/schemas/validation.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any

class ConflictCandidate(BaseModel):
    candidate_id: str
    value: str
    unit: Optional[str] = None
    source_name: str
    source_type: str  # 'pdf', 'website', 'catalog', 'image', 'third_party'
    source_url: Optional[str] = None
    page_number: Optional[int] = None
    confidence: float
    evidence_text: str

class ConflictItem(BaseModel):
    id: str
    product_id: str
    attribute_name: str
    key: str
    current_value: Optional[str] = None
    status: str  # 'conflict', 'needs_review', 'resolved'
    candidates: List[ConflictCandidate]
    created_at: str

class ReviewActionRequest(BaseModel):
    action: str  # 'select_candidate', 'edit_manual', 'mark_na', 'reject_ai'
    selected_candidate_id: Optional[str] = None
    manual_value: Optional[str] = None
    manual_unit: Optional[str] = None
    reviewer_name: str = "Lead Quality Engineer"
    notes: Optional[str] = None

class ReviewHistoryItem(BaseModel):
    id: str
    product_id: str
    attribute_name: str
    key: str
    previous_value: Optional[str] = None
    final_value: str
    reviewer: str
    action: str
    timestamp: str

class ValidationSummary(BaseModel):
    product_id: str
    total_attributes_validated: int
    matching_specs_count: int
    conflicts_count: int
    low_confidence_count: int
    unverified_enriched_count: int
    overall_confidence: float
    confidence_tier: str  # 'High Confidence', 'Medium Confidence', 'Low Confidence'
    validation_status: str  # 'verified', 'human_verified', 'needs_review', 'conflict'
